main/autogen/data/definitions.cc

```diff
@@ -171,12 +171,29 @@ vector<string> ParsedFile::listAllClasses(core::Context ctx) {
 
 // Convert this parsedfile to a msgpack representation
 string ParsedFile::toMsgpack(core::Context ctx, int version, const AutogenConfig &autogenCfg) {
-    MsgpackWriter write(version);
-    return write.pack(ctx, *this, autogenCfg);
+    int version_bit_8 = version >> 7; // bits higher than position 8 should never be set.
+    int version_low_7 = (version & 127);
+
+    if (version_bit_8 == 1) {
+        MsgpackWriterLite write(version_low_7);
+
+        return write.pack(ctx, *this, autogenCfg);
+    } else {
+        MsgpackWriter write(version_low_7);
+
+        return write.pack(ctx, *this, autogenCfg);
+    }
```

Comment:
Is the intent here that the external invoker of depdb passes a version with the high bit set?  a) that seems like putting the responsibility in the wrong place; b) our `MAX_VERSION` constant would need to be updated for that (maybe you did it this way because you ran into `MAX_VERSION` issues in the first place?); and c) even if this bit-passing is intended to live somewhere under `main/autogen/`, I don't see that being done in this PR.

This all seems more complicated than just having `MsgpackWriterLite::msgpackGlobalHeader` tack on the high bit itself.

Reply:
This works because the assertion for `MAX_VERSION` runs at the point of instantiating the `MsgpackWriter` or `Lite` object, which still gets the version with the high bit stripped out.

We need some way of communicating via the cli, either a flag like I had before or this: I figure since we're already passing version in the cli this is reasonable to do.